core/src/main/scala/libretto/CoreLib.scala

```diff
@@ -2830,40 +2830,37 @@ class CoreLib[DSL <: CoreDSL](val dsl: DSL) { lib =>
     ): LList1[A] -⚬ (Unlimited[A |*| Ā] |*| LList1[A]) =
       unfold(LList1.borrow(lInvert))
 
-    implicit def comonoidUnlimited[A]: Comonoid[Unlimited[A]] =
-      new Comonoid[Unlimited[A]] {
-        def counit : Unlimited[A] -⚬ One                             = Unlimited.discard
-        def split  : Unlimited[A] -⚬ (Unlimited[A] |*| Unlimited[A]) = Unlimited.split
-      }
+    given comonoidUnlimited[A]: Comonoid[Unlimited[A]] with {
```

Comment:
This implicit stuff is really used a lot explicitly…

Reply:
How would you otherwise solve this problem:

```scala
class Foo

given Option[Foo] = None

opaque type Bar = Foo

given Option[Bar] = summon[Option[Foo]] // Ambiguous given instances: both given instance given_Option_Foo
                                        // [...] and given instance given_Option_Bar [...] match type Option[Foo]
```

https://scastie.scala-lang.org/vn711zElRIiWrSE37oAOlw